pkg/controller.v1/tensorflow/tfjob_controller.go

```diff
@@ -21,55 +21,90 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+	volcanoclient "volcano.sh/apis/pkg/client/clientset/versioned"
+)
 
-	tensorflowv1 "github.com/kubeflow/tf-operator/pkg/apis/tensorflow/v1"
-	tfv1 "github.com/kubeflow/tf-operator/pkg/apis/tensorflow/v1"
-	trainingoperatorcommon "github.com/kubeflow/tf-operator/pkg/common"
-	"github.com/kubeflow/tf-operator/pkg/common/util"
+const (
+	// tfJobCreatedReason is added in a tfjob when it is created.
+	tfJobCreatedReason = "TFJobCreated"
+	// tfJobSucceededReason is added in a tfjob when it is succeeded.
+	tfJobSucceededReason = "TFJobSucceeded"
+	// tfJobRunningReason is added in a tfjob when it is running.
+	tfJobRunningReason = "TFJobRunning"
+	// tfJobFailedReason is added in a tfjob when it is failed.
+	tfJobFailedReason = "TFJobFailed"
+	// tfJobRestarting is added in a tfjob when it is restarting.
+	tfJobRestartingReason = "TFJobRestarting"
+
+	failedMarshalTFJobReason  = "InvalidTFJobSpec"
+	FailedDeleteJobReason     = "FailedDeleteJob"
+	SuccessfulDeleteJobReason = "SuccessfulDeleteJob"
+
+	controllerName = "tf-operator"
+
+	// labels for pods and servers.
+	tfReplicaTypeLabel  = "replica-type"
+	tfReplicaIndexLabel = "replica-index"
+	labelGroupName      = "group-name"
+	// Deprecated label for backwards compatibility. Has to be removed
+	labelTFJobName = "tf-job-name"
+	// volcanoTaskSpecKey task spec key used in pod annotation when EnableGangScheduling is true
+	volcanoTaskSpecKey = "volcano.sh/task-spec"
+
+	// gang scheduler name.
+	gangSchedulerName = "volcano"
+	// tfConfig is the environment variable name of TensorFlow cluster spec.
+	tfConfig = "TF_CONFIG"
+	// exitedWithCodeReason is the normal reason when the pod is exited because of the exit code.
+	exitedWithCodeReason = "ExitedWithCode"
+	// podTemplateRestartPolicyReason is the warning reason when the restart
+	// policy is set in pod template.
+	podTemplateRestartPolicyReason = "SettedPodTemplateRestartPolicy"
+	// podTemplateSchedulerNameReason is the warning reason when other scheduler name is set
+	// in pod templates with gang-scheduling enabled
+	podTemplateSchedulerNameReason = "SettedPodTemplateSchedulerName"
+	// gangSchedulingPodGroupAnnotation is the annotation key used by batch schedulers
+	gangSchedulingPodGroupAnnotation = "scheduling.k8s.io/group-name"
 )
 
 var (
-	defaultCleanPodPolicy = commonv1.CleanPodPolicyNone
+	// KeyFunc is the short name to DeletionHandlingMetaNamespaceKeyFunc.
+	// IndexerInformer uses a delta queue, therefore for deletes we have to use this
+	// key function but it should be just fine for non delete events.
+	KeyFunc = cache.DeletionHandlingMetaNamespaceKeyFunc
```

Comment:
Please check if it's same to `common.KeyFunc` ?

Reply:
indeed. can just use that directly.